test/sharness/lib/test-lib.sh

```diff
@@ -204,6 +183,22 @@ test_launch_ipfs_daemon() {
 		ipfs daemon $args >actual_daemon 2>daemon_err &
 	'
 
+	# wait for api file to show up
+	test_expect_success "api file shows up" '
+		test_wait_for_file 20 100ms "$IPFS_PATH/api" &&
+		ADDR_API=$(cat "$IPFS_PATH/api") &&
+		PORT_API=$(port_from_maddr $ADDR_API) &&
+		ADDR_GWAY=$(sed -n "s/^Gateway (.*) server listening on //p" actual_daemon) &&
+		PORT_GWAY=$(port_from_maddr $ADDR_GWAY)
+	'
+
+
+	test_expect_success "set swarm address vars" '
+		ipfs swarm addrs local > addrs_out &&
+		ADDR_SWARM=$(grep "127.0.0.1" addrs_out) &&
+		PORT_SWARM=$(port_from_maddr $ADDR_SWARM)
```

Comment:
we may want to do:
```
MADDR_API=$(grep "127.0.0.1" addrs_out) &&
ADDR_API=$(convert_tcp_maddr $MADDR_API) &&

MADDR_SWARM=$(grep "127.0.0.1" addrs_out) &&
ADDR_SWARM=$(convert_tcp_maddr $MADDR_SWARM) &&
```
to avoid having to call the function everywhere (less typing).

also we maybe should invert these to:

```
$API_MADDR
$API_ADDR
$API_PORT

$SWARM_MADDR
$SWARM_ADDR
$SWARM_PORT
```
and improve readability / understanding for people writing tests.